extractor/src/test/java/org/schabi/newpipe/extractor/services/media_ccc/MediaCCCStreamExtractorTest.java

```diff
@@ -6,96 +6,182 @@
-        assertEquals(2, extractor.getAudioStreams().size());
-    }
-
-    @Test
-    public void testGetTextualUploadDate() throws ParsingException {
-        Assert.assertEquals("2018-05-11T02:00:00.000+02:00", extractor.getTextualUploadDate());
+public class MediaCCCStreamExtractorTest {
+
+    public static class Gpn18Tmux {
+        private static MediaCCCStreamExtractor extractor;
+
+        @BeforeClass
+        public static void setUpClass() throws Exception {
+            NewPipe.init(DownloaderTestImpl.getInstance());
+
+            extractor = (MediaCCCStreamExtractor) MediaCCC.getStreamExtractor("https://media.ccc.de/v/gpn18-105-tmux-warum-ein-schwarzes-fenster-am-bildschirm-reicht");
+            extractor.fetchPage();
+        }
+
+        @Test
+        public void testServiceId() throws Exception {
+            assertEquals(2, extractor.getServiceId());
+        }
+
+        @Test
+        public void testName() throws Exception {
+            assertEquals("tmux - Warum ein schwarzes Fenster am Bildschirm reicht", extractor.getName());
+        }
+
+        @Test
+        public void testId() throws Exception {
+            assertEquals("gpn18-105-tmux-warum-ein-schwarzes-fenster-am-bildschirm-reicht", extractor.getId());
+        }
+
+        @Test
+        public void testUrl() throws Exception {
+            assertIsSecureUrl(extractor.getUrl());
+            assertEquals("https://api.media.ccc.de/public/events/gpn18-105-tmux-warum-ein-schwarzes-fenster-am-bildschirm-reicht", extractor.getUrl());
+        }
+
+        @Test
+        public void testOriginalUrl() throws Exception {
+            assertIsSecureUrl(extractor.getOriginalUrl());
+            assertEquals("https://media.ccc.de/v/gpn18-105-tmux-warum-ein-schwarzes-fenster-am-bildschirm-reicht", extractor.getOriginalUrl());
+        }
+
+        @Test
+        public void testThumbnail() throws Exception {
+            assertIsSecureUrl(extractor.getThumbnailUrl());
+            assertEquals("https://static.media.ccc.de/media/events/gpn/gpn18/105-hd.jpg", extractor.getThumbnailUrl());
+        }
+
+        @Test
+        public void testUploaderName() throws Exception {
+            assertEquals("gpn18", extractor.getUploaderName());
+        }
+
+        @Test
+        public void testUploaderUrl() throws Exception {
+            assertIsSecureUrl(extractor.getUploaderUrl());
+            assertEquals("https://api.media.ccc.de/public/conferences/gpn18", extractor.getUploaderUrl());
+        }
+
+        @Test
+        public void testUploaderAvatarUrl() throws Exception {
+            assertIsSecureUrl(extractor.getUploaderAvatarUrl());
+            assertEquals("https://static.media.ccc.de/media/events/gpn/gpn18/logo.png", extractor.getUploaderAvatarUrl());
+        }
+
+        @Test
+        public void testVideoStreams() throws Exception {
+            assertEquals(4, extractor.getVideoStreams().size());
+        }
```

Comment:
Please do something like this (also below and in all MediaCCC testVideoStreams and testAudioStreams functions).
The below code is untested
```suggestion
        public void testVideoStreams() throws Exception {
            List<VideoStream> streams = extractor.getVideoStreams();
            assertEquals(4, streams.size());
            for (VideoStream stream : streams) {
                assertIsSecureUrl(stream.getUrl())
            }
        }
```